src/main/java/jenkins/plugins/openstack/compute/JCloudsCloud.java

```diff
@@ -386,28 +386,39 @@ public void doProvision(
             return;
         }
 
-        CloudStatistics.ProvisioningListener provisioningListener = CloudStatistics.ProvisioningListener.get();
-        ProvisioningActivity.Id id = new ProvisioningActivity.Id(this.name, t.name);
-
         JCloudsSlave node;
         try {
-            provisioningListener.onStarted(id);
-            node = t.provisionSlave(this, id);
-            provisioningListener.onComplete(id, node);
+            node = doProvisionSlave(t);
         } catch (Openstack.ActionFailed ex) {
-            provisioningListener.onFailure(id, ex);
             req.setAttribute("message", ex.getMessage());
             req.setAttribute("exception", ex);
             rsp.forward(this,"error",req);
             return;
-        } catch (Throwable ex) {
-            provisioningListener.onFailure(id, ex);
-            throw ex;
         }
-        Jenkins.getActiveInstance().addNode(node);
         rsp.sendRedirect2(req.getContextPath() + "/computer/" + node.getNodeName());
     }
 
+   @Restricted(NoExternalUse.class)
+   public @Nonnull JCloudsSlave doProvisionSlave(JCloudsSlaveTemplate template) throws IOException, Openstack.ActionFailed{
```

Comment:
Since this is not a Stapler action method and is in fact invoked from code, there is no reason to stick to `doXxx` name.

Reply:
Good point. Fixed.